pre_commit/commands/run.py

```diff
@@ -157,12 +157,14 @@ def _run_single_hook(
         retcode = 0
         diff_after = diff_before
         files_modified = False
+        hook_failed = False
         out = b''
     elif not filenames and not hook.always_run:
         duration = None
         retcode = 0
         diff_after = diff_before
         files_modified = False
+        hook_failed = False
         out = b''
```

Comment:
It seems like all of these vars are immediately overwritten in the else branch so pulling these out to always 'reset' them wouldn't be too bad. Minor optimization. Unless our plan is to keep refactors to a minimum so that merging in future upstream releases has minimal conflicts.